auth/router.py

```
@@ -48,21 +50,24 @@ async def create_user(user: UserCreate, db: AsyncSession = Depends(get_db)):
         db.add(db_user)
         await db.commit()
         await db.refresh(db_user)
+        logger.info("User registered successfully.")
         return db_user
     except ValidationError as e:
         # Handle validation errors from Pydantic
+        logger.error(f"Validation error: {e.errors()}")
         raise HTTPException(
             status_code=status.HTTP_422_UNPROCESSABLE_ENTITY,
             detail=e.errors()
         )
```

Ruff

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)